drivers/net/nsos_adapt.c

```diff
@@ -130,6 +138,9 @@ static int socket_proto_from_nsos_mid(int proto_mid, int *proto)
 	case NSOS_MID_IPPROTO_RAW:
 		*proto = IPPROTO_RAW;
 		break;
+	case NSOS_MID_IPPROTO_ETH_P_ALL:
+		*proto = htons(ETH_P_ALL);
```

Comment:
Why is `htons()` used only for that (ETH_P_ALL) option?

Reply:
See commit 99693bee5f2762f84b78ed90740753cdec40f394 for details.

Reply:
I've used `htons()` here due to:
[socket_API](https://docs.zephyrproject.org/latest/releases/migration-guide-3.7.html):

> The packet socket (type AF_PACKET) protocol field in socket API call has changed. The protocol field should be in network byte order so that we are compatible with Linux socket calls. Linux expects the protocol field to be htons(ETH_P_ALL) if it is desired to receive all the network packets. 

So, for example now, what I have is `nsos_socket_create(htons(ETH_P_ALL))` as input  (htons(3)), followed by:
- `socket_proto_to_nsos_mid()`, proto_mid is computed to 3.
- then `nsos_adapt_socket() `with proto_mid=3 is called.
   - Here `socket_proto_from_nsos_mid()` computes `proto = htons(NSOS_MID_IPPROTO_ETH_P_ALL) = htons(3) `which is the same as htons(ETH_P_ALL).
   - Then `socket()` is called with `proto = htons(ETH_P_ALL)`.

To avoid this, I should have defined NSOS_MID_IPPROTO_ETH_P_ALL to htons(ETH_P_ALL) and modified other functions. But I've defined NSOS_MID_IPPROTO_ETH_P_ALL to 3 to keep the same define value as IPROTO_ETH_P_ALL.